src/components/SideTray/SideTray.tsx

```diff
@@ -7,7 +7,65 @@ import { parseCssSize } from '../../shared/css-utils';
 // This component's dependencies
 import styles from './SideTray.module.css';
 
-const SideTray = ({
+interface SideTrayProps {
+  children: ReactNode;
+  className?: string;
+  direction?: string;
+  height?: string | number;
+  onClose: (...args) => void;
+  style?: object;
+  visible?: boolean;
+  width?: string | number;
+}
+
+interface SideTrayStyles {
+  bottom: number | string;
+  height: string;
+  left: number | string;
+  right: number | string;
+  top: number | string;
+  width: string;
+}
+
+interface ChildComponentsProps {
+  children: ReactNode;
+  className?: string;
+}
+
+const Header: React.FC<ChildComponentsProps> = ({ className, ...props }) => (
+  <div {...props} className={`${styles.header} ${className}`} />
+);
+Header.propTypes = {
+  children: PropTypes.node.isRequired,
+  className: PropTypes.string,
+};
+Header.defaultProps = { className: '' };
+
+const Footer: React.FC<ChildComponentsProps> = ({ className, ...props }) => (
```

Comment:
When we switch to TS, we generally remove the defaultProps and declare default function parameters instead.
Badge.tsx is a pretty good reference, iirc.